astropy/coordinates/representation.py

```diff
@@ -42,6 +42,24 @@
 REPRESENTATION_CLASSES = {}
 DIFFERENTIAL_CLASSES = {}
 
+# a hash for the content of the above two dicts, cached for speed.
+_REPRDIFF_HASH = None
+def get_reprdiff_cls_hash():
+    """
+    Returns a hash value that should be invariable if the
+    `REPRESENTATION_CLASSES` and `DIFFERENTIAL_CLASSES` dictionaries have not
+    changed.
+    """
+    global _REPRDIFF_HASH
```

Comment:
I acknowledge your point here.  I considered something more drastic like combining `REPRESENTATION_CLASSES` and `DIFFERENTIAL_CLASSES` into one thing and then storing the hash there... but that ended up *more* complicated.  It seemed like an acceptable compromise because those two dictionaries already live here, but I share your "a bit uncomfortable".